optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bavet/uni/BavetAbstractUniConstraintStream.java

```diff
@@ -90,8 +91,23 @@ public void addChildStream(BavetAbstractUniConstraintStream<Solution_, A> childS
         }
         if (!(joiner instanceof AbstractBiJoiner)) {
             throw new IllegalArgumentException("The joiner class (" + joiner.getClass() + ") is not supported.");
+        } else if (joiner instanceof FilteringBiJoiner) {
+            return join(otherStream)
+                    .filter(((FilteringBiJoiner<A, B>) joiner).getFilter());
```

Comment:
Who's responsible for creaming off the FilteringBiJoiner? Is it UniConstraintStream or the Drools/BavetAbstractUniConstraintStream? currently I see code doing this in both sides.

Can we extra some of this into seperate method?

Reply:
Unfortunately, this needs to happen here. This method could be called directly (and not through the other `join()` methods), and therefore this joiner could be a `FilteringBiJoiner`. There is no other case where that could happen.

Reply:
good point. Maybe we need to introduce a AbstractUniConstraintStream to capture code that is common between Drools and Bavet to avoid duplicating that code, without exposing it in the interface? Similar to how AbstractScoreDirector sits between ScoreDirector and DroolsScoreDirector/etc?
We could make an innerJoin() method that the implementations need to implement.

Not sure if this is worth it's weight though?